lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -71,7 +71,11 @@ def paintEvent(self, e):
         In Qt, all drawing should be done inside of here when a widget is
         shown onscreen.
         """
-        FigureCanvasAgg.draw(self)
+        if not hasattr(self, "_agg_redraw_flag"):
+            self._agg_redraw_flag = True
+        if self._agg_redraw_flag:
+            self._agg_redraw_flag = False
```

Comment:
I would have, but I'm still confused about these draw_idle callbacks from within the draw call. If there is a reason for those you cannot clear the flag after draw since it might have been re-set during draw on purpose.